Sources/SPTDataLoaderService.m

```diff
@@ -52,6 +54,10 @@ @interface SPTDataLoaderService () <SPTDataLoaderRequestResponseHandlerDelegate,
 @end
 
 @implementation SPTDataLoaderService
+{
+    NSURLSession *_nonWaitingSession;
+    NSURLSession *_waitingSession;
```

Comment:
The `sessionForRequest:` mechanism could be extended to handle other session configuration flags, such as [`allowsCellularAccess`](https://developer.apple.com/documentation/foundation/nsurlsessionconfiguration/1409406-allowscellularaccess?language=objc), [`allowsExpensiveNetworkAccess`](https://developer.apple.com/documentation/foundation/nsurlsessionconfiguration/3235752-allowsexpensivenetworkaccess?language=objc), and [`allowsConstrainedNetworkAccess`](https://developer.apple.com/documentation/foundation/nsurlsessionconfiguration/3235751-allowsconstrainednetworkaccess?language=objc) (iOS 13 Low Data Mode).

If we do this, I’d suggest merging the flags into a bitfield, masking out the ones that aren’t available on the current OS, and using that as a key into a dictionary of `NSURLSession`s.